File: sdk/kotlin/src/main/kotlin/vn/hitechcloud/sdk/Collocation.kt
package vn.hitechcloud.sdk

/** Collocation Services: PDU port management */
class CollocationResource(private val httpClient: HttpClient) {
    /** GET /api/service/{id}/pdu - List PDU ports assigned to service */
    fun listPduPorts(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/service/{id}/pdu", params)
    }

    /** GET /api/service/{id}/pdu/{port} - Get PDU port/outlet state */
    fun getPduPortState(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/service/{id}/pdu/{port}", params)
    }

    /** POST /api/service/{id}/pdu/{port} - Set PDU port/outlet state */
    fun setPduPortStatus(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.post("/api/service/{id}/pdu/{port}", params)
    }

}

File: sdk/swift/Sources/HiTechCloudSDK/Collocation.swift
import Foundation

/// Collocation Services: PDU port management
public class CollocationResource {
    private let httpClient: HTTPClient
    public init(httpClient: HTTPClient) { self.httpClient = httpClient }

    /// GET /api/service/{id}/pdu - List PDU ports assigned to service
    public func listPduPorts(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/service/{id}/pdu", params: params)
    }

    /// GET /api/service/{id}/pdu/{port} - Get PDU port/outlet state
    public func getPduPortState(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/service/{id}/pdu/{port}", params: params)
    }

    /// POST /api/service/{id}/pdu/{port} - Set PDU port/outlet state
    public func setPduPortStatus(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.post("/api/service/{id}/pdu/{port}", params: params)
    }

}
